### ISSUE TITLE
Crash upon accepting voice call

### ISSUE THREAD

**Issue by opusforlife2** [vector-im/element-android#5421]
### Steps to reproduce

Element completely crashed when I accepted a voice call just now. I don't know if it's reproducible yet.

Specifically, the crash occurred when the call status changed to "Call connecting..."

Logs:
```
FATAL EXCEPTION: LIVE_ENTITY_BACKGROUND
Process: im.vector.app, PID: 12473
java.util.ConcurrentModificationException
	at java.util.ArrayList$Itr.next(ArrayList.java:860)
	at org.matrix.android.sdk.internal.session.call.CallEventProcessor.onPostProcess(CallEventProcessor.kt:2)
	at org.matrix.android.sdk.internal.database.EventInsertLiveObserver$onChange$2.invokeSuspend(EventInsertLiveObserver.kt:11)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:3)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:18)
	at android.os.Handler.handleCallback(Handler.java:883)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loop(Looper.java:214)
	at android.os.HandlerThread.run(HandlerThread.java:67)
```

#### What did you expect?

For the voice call to begin normally.

And for Element to offer to send a crash report upon restart.

#### What happened instead?

Neither. Element just crashed. The logs I've caught were through Scoop, so I've selected "No" for logs.

### Your phone model

Xiaomi Mi 5

### Operating system version

Android 10 (LOS 17.1)

### Application version and app store

1.4.2 (F-Droid)

### Will you send logs?

No

---- 

EDIT by @ouchadam 

We're also getting reports in the play store console with the exact same stacktrace 

| Occurrences (30 days) | Total Occurrences | Impacted users (lifetime) |
| --- | --- | --- |
| 143 | 149 | 125 |

https://play.google.com/console/u/0/developers/{redacted}/app/{redacted}/vitals/crashes/3c579fd0/details?installedFrom=PLAY_STORE&days=30

**Comment by opusforlife2** [comment]
Hello triage team.

**Comment by ouchadam** [comment]
marked as uncommon whilst there are any other reports, `ConcurrentModificationException` will be tricky to reproduce but we can definitely guard against it

it also looks like the backing list is accessed from multiple threads https://github.com/vector-im/element-android/blob/develop/matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/call/CallEventProcessor.kt

**Comment by ouchadam** [comment]
I've found the same crash in the play store console, have edited the issue description to include the current stats

**Comment by ouchadam** [comment]
fixed in 1.4.14
